src/NServiceBus.Persistence.AzureStorage/Subscriptions/AzureSubscriptionStorage.cs

```diff
@@ -142,5 +190,11 @@ public int GetHashCode(Subscriber obj)
                 return (obj.Endpoint ?? string.Empty).Length;
             }
         }
+
+        internal class CacheItem
+        {
+            public DateTime Stored;
+            public Task<IEnumerable<Subscriber>> Subscribers;
```

Comment:
What @SimonCropp summarized. In this case there is no state machine involved and therefore no allocs beside the task. In the future this might be a good opportunity to use a ValueTask.